evm_loader/lib/src/tracing/tracers/openeth/tracer.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt::Debug;
+
+use serde_json::Value;
+
+use evm_loader::evm::tracing::{EmulationResult, Event, EventListener};
+use evm_loader::types::hexbytes::HexBytes;
+
+use crate::tracing::tracers::openeth::types::{CallAnalytics, TraceResults};
+
+#[derive(Debug)]
+pub struct OpenEthereumTracer {
+    output: Option<HexBytes>,
+    call_analytics: CallAnalytics,
+}
+
+impl OpenEthereumTracer {
+    #[must_use]
+    pub fn new(call_analytics: CallAnalytics) -> OpenEthereumTracer {
+        OpenEthereumTracer {
+            output: None,
+            call_analytics,
+        }
+    }
+}
+
+impl EventListener for OpenEthereumTracer {
+    fn event(&mut self, event: Event) {
+        if let Event::EndStep {
+            gas_used: _gas_used,
+            return_data,
+        } = event
+        {
+            self.output = return_data.map(Into::into);
+        }
+    }
+
+    fn into_traces(self: Box<Self>, emulation_result: EmulationResult) -> Value {
+        serde_json::to_value(TraceResults {
+            output: self.output.unwrap_or_default(),
+            trace: vec![],
+            vm_trace: None,
+            state_diff: if self.call_analytics.state_diffing {
+                Some(emulation_result.state_diff)
+            } else {
+                None
+            },
+        })
+        .unwrap()
```

Comment:
Could we consider altering it to try_into_traces() that returns a Result? That could potentially eliminate any panic situation, couldn't it?

Reply:
The same happens in https://github.com/neonlabsorg/neon-evm/blob/NDEV-2301-slim/evm_loader/lib/src/tracing/tracers/struct_logger.rs#L220. The doc string of `serde_json::to_value` also states that:
```Rust
/// # Errors
///
/// This conversion can fail if `T`'s implementation of `Serialize` decides to
/// fail, or if `T` contains a map with non-string keys.
```
The serialization should never fail and even if it does, it will only fail the current tracing request.